eval/compile.rkt

```diff
@@ -270,8 +270,10 @@
   (define registers (make-vector register-count))
   (define repeats (make-vector register-count #f)) ; flags whether an op should be evaluated
   (define precisions (make-vector register-count)) ; vector that stores working precisions
-  ;; starting precisions for the first, un-tuned iteration
-  (define initial-precisions (setup-vstart-precs instructions (length vars) roots discs))
+  ;; vector for adjusting precisions
+  (define incremental-precisions (setup-vstart-precs instructions (length vars) roots discs))
+  (define initial-precision
+    (+ (argmax identity (map discretization-target discs)) (*base-tuning-precision*)))
```

Comment:
I mean I think this is fine but I guess ideally we'd use some more clever algorithm that's aware of different precision needs for different roots?

Reply:
We can, I can create a new vector that will store precisions only for the first iteration.
As for now I have:
1) Precision vector `vprecs` that I use when executing
2) Precision (int) that I feel `vprecs` with for the first iteration
3) And _incremental precisions_ which is a vector with `+5` weird propogation that we like:) Used in `backward-pass`

Just instead of the 2nd bullet point we can have another vector